src/controllers/userController.js

```diff
@@ -13,6 +20,11 @@ const getSingleUser = async(req,res) => { // Function to get a single user by ID
     res.status(StatusCodes.OK).json({ user }); // Send a JSON response with the status code 200 OK and the user
 };
 
+const showCurrentUser = async(req,res) => { // Function to get the current user's information
+    // Send a JSON response with the status code 200 OK and the current user's information
+    res.status(StatusCodes.OK).json({ user: req.user });
```

Comment:
One thing that will be a bit odd here is that this will return a tokenized user (with `userId,` `role`, `email`) but not much other info about the current user. If the frontend wants to use this endpoint for something like showing the user profile page, they won't have the user's name or credit card info for example. Instead, lets ensure that all endpoints in the `userController` return a full user (minus the password of course). 

For the `authController` it's fine to return the tokenized user since we shouldn't expect the frontend to use the return value of those endpoints for the same type of purpose. 

Here, this would probably mean that you need to add a `User.findOne` call to this handler

Reply:
Thank you very much for the review. I may have misunderstood, so I would like to clarify. Does my error handling and data return in the showCurrentUser function look correct? Are there any other improvements you would recommend?